scaaml_intro/train.py

```diff
@@ -77,17 +79,23 @@ def train_model(config):
             # multi gpu
             strategy = tf.distribute.MirroredStrategy()
             with strategy.scope():
-                model = get_model(input_shape, attack_point, config)
-
-                # model recording setup
                 stub = get_model_stub(attack_point, attack_byte, config)
+                file_path = f"models/{stub}.keras"
```

Comment:
This modernises the code. However we would be silently losing backwards compatibility with pretrained models. Since this code is for the 2019 Defcon presentation I would prefer to pin a TensorFlow version here (say 2.12) and have more up to date code for the more recent projects. What do you think?

Reply:
you're right back compatibility is a concern but when I tried to run the code it threw an error asking to add a keras extension so I did. I was using the latest version of everything.

Reply:
ok this might not be an issue if we install the version mentioned in requirements.

Reply:
it wasn't loading with no extension back ward compatibility is a concern but